scapy/contrib/loraphy2wan.py

```diff
@@ -0,0 +1,681 @@
+# along with Scapy. If not, see <http://www.gnu.org/licenses/>.
+
+# scapy.contrib.description = LoRa PHY to WAN Layer
+# scapy.contrib.status = loads
+
+
+"""
+    Copyright (C) 2020 Sebastien Dudek (@FlUxIuS @PentHertz)
+"""
+
+from __future__ import absolute_import
+
+from scapy.packet import Packet
+from scapy.fields import BitField, ByteEnumField, ByteField, \
+    ConditionalField, IntField, LEShortField, PacketListField, \
+    StrFixedLenField, X3BytesField, XByteField, XIntField, \
+    XShortField, BitFieldLenField, LEX3BytesField, XBitField, \
+    BitEnumField, XLEIntField, StrField
+
+
+class FCtrl_DownLink(Packet):
+    name = "FCtrl_DownLink"
+    fields_desc = [BitField("ADR", 0, 1),
+                   BitField("ADRACKReq", 0, 1),
+                   BitField("ACK", 0, 1),
+                   BitField("FPending", 0, 1),
+                   BitFieldLenField("FOptsLen", 0, 4)]
+
+    # pylint: disable=R0201
+    def extract_padding(self, p):
+        return "", p
+
+
+class FCtrl_UpLink(Packet):
+    name = "FCtrl_UpLink"
+    fields_desc = [BitField("ADR", 0, 1),
+                   BitField("ADRACKReq", 0, 1),
+                   BitField("ACK", 0, 1),
+                   BitField("ClassB", 0, 1),
+                   BitFieldLenField("FOptsLen", 0, 4)]
+
+    # pylint: disable=R0201
+    def extract_padding(self, p):
+        return "", p
+
+
+class DevAddrElem(Packet):
+    name = "DevAddrElem"
+    fields_desc = [XByteField("NwkID", 0x0),
+                   LEX3BytesField("NwkAddr", b"\x00" * 3)]
+
+
+CIDs_up = {0x01: "ResetInd",
+           0x02: "LinkCheckReq",
+           0x03: "LinkADRReq",
+           0x04: "DutyCycleReq",
+           0x05: "RXParamSetupReq",
+           0x06: "DevStatusReq",
+           0x07: "NewChannelReq",
+           0x08: "RXTimingSetupReq",
+           0x09: "TxParamSetupReq",  # LoRa 1.1 specs
+           0x0A: "DlChannelReq",
+           0x0B: "RekeyInd",
+           0x0C: "ADRParamSetupReq",
+           0x0D: "DeviceTimeReq",
+           0x0E: "ForceRejoinReq",
+           0x0F: "RejoinParamSetupReq"}  # end of LoRa 1.1 specs
+
+
+CIDs_down = {0x01: "ResetConf",
+             0x02: "LinkCheckAns",
+             0x03: "LinkADRAns",
+             0x04: "DutyCycleAns",
+             0x05: "RXParamSetupAns",
+             0x06: "DevStatusAns",
+             0x07: "NewChannelAns",
+             0x08: "RXTimingSetupAns",
+             0x09: "TxParamSetupAns",  # LoRa 1.1 specs here
+             0x0A: "DlChannelAns",
+             0x0B: "RekeyConf",
+             0x0C: "ADRParamSetupAns",
+             0x0D: "DeviceTimeAns",
+             0x0F: "RejoinParamSetupAns"}  # end of LoRa 1.1 specs
+
+
+class ResetInd(Packet):
+    name = "ResetInd"
+    fields_desc = [ByteField("Dev_version", 0)]
+
+
+class ResetConf(Packet):
+    name = "ResetConf"
+    fields_desc = [ByteField("Serv_version", 0)]
+
+
+class LinkCheckReq(Packet):
+    name = "LinkCheckReq"
+    fields_desc = []
```

Comment:
You can remove this line as `fields_desc`  defaults to an empty list.

Reply:
done